### PR TITLE
[MOB-791] Implemented page view events to Rakam;

### PR DESCRIPTION
**Pull request by franciscoCalado**
**What does this PR do?**

This PR implements page view events to Rakam across the app;

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] BasePageViewFragment.kt

**How should this be manually tested?**

Check all screens you can think of in the app and verify the event is being sent. You can filter logcat by "rakam" to see the specific event logger that logs these events.

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-791](https://aptoide.atlassian.net/browse/MOB-791)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/App.kt [hunk 1]
```diff
@@ -152,6 +156,8 @@ class App : MultiDexApplication(), HasAndroidInjector, BillingDependenciesProvid
     }
   }
 
+  fun analyticsManager() = analyticsManager
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/App.kt:159]: Not being used

File: app/src/main/java/com/asfoundation/wallet/referrals/ReferralsFragment.kt [hunk 2]
```diff
@@ -9,16 +9,16 @@ import android.view.ViewGroup
...
-import dagger.android.support.DaggerFragment
 import io.reactivex.disposables.CompositeDisposable
 import kotlinx.android.synthetic.main.invited_friends_animation_list.*
 import kotlinx.android.synthetic.main.referrals_layout.*
 import java.math.BigDecimal
 import javax.inject.Inject
 import kotlin.math.roundToInt
 
-class ReferralsFragment : DaggerFragment(), ReferralsView {
+class ReferralsFragment : BasePageViewFragment(), ReferralsView {
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/referrals/ReferralsFragment.kt]: This is a bottom sheet fragment having this here will send 2 events for the same fragment (This appears at the same time as the InviteFriendsFragment

File: app/src/main/java/com/asfoundation/wallet/ui/gamification/HowItWorksFragment.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ import java.text.SimpleDateFormat
 import java.util.*
 import javax.inject.Inject
 
-class HowItWorksFragment : DaggerFragment(), HowItWorksView {
+class HowItWorksFragment : BasePageViewFragment(), HowItWorksView {
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/gamification/HowItWorksFragment.kt]: Bottom sheet of the LegacyGamificationFragment, can be removed

**Review by JDSM01 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/App.kt (hunk 1 above)
Comment (franciscoCalado) [reply app/src/main/java/com/asfoundation/wallet/App.kt:159]: it is, don't know why IDE says otherwise (check baseFragment)

**Pull request merged**